library/src/main/java/dev/chrisbanes/insetter/Insetter.java

```diff
@@ -332,6 +334,89 @@ public static void setEdgeToEdgeSystemUiFlags(@NonNull final View view, final bo
+   */
+  public static void applyInsetsWhenKeyboardDisplayed(@NonNull final View v,
+                                                      final boolean marginBottomForKeyboard,
+                                                      final boolean paddingBottomForKeyboard) {
+    setOnApplyInsetsListener(v, new OnApplyInsetsListener() {
+      @Override
+      public void onApplyInsets(@NonNull View view, @NonNull WindowInsetsCompat insets, @NonNull ViewState initialState) {
+        boolean keyboardOpen = isKeyboardOpen(view);
+
+        int paddingBottom;
+        if (keyboardOpen) {
+          paddingBottom = insets.getSystemWindowInsetBottom();
+        } else {
+          paddingBottom = initialState.getPaddings().getBottom();
+        }
+
+        int marginBottom;
+        if (keyboardOpen) {
+          marginBottom = insets.getSystemWindowInsetBottom();
+        } else {
+          marginBottom = initialState.getMargins().getBottom();
+        }
+
+        if (marginBottomForKeyboard) {
+          final ViewGroup.LayoutParams lp = view.getLayoutParams();
+          if (lp instanceof ViewGroup.MarginLayoutParams) {
+            final ViewGroup.MarginLayoutParams mlp = (ViewGroup.MarginLayoutParams) lp;
+
+            if (mlp.bottomMargin != marginBottom) {
+              mlp.bottomMargin = marginBottom;
+              view.setLayoutParams(lp);
+
+              if (Log.isLoggable(TAG, Log.DEBUG)) {
+                Log.d(
+                        TAG,
+                        String.format(
+                                Locale.US,
+                                "applyInsetsWhenKeyboardDisplayed. Applied margin to %s: bottom=%d}",
+                                view,
+                                marginBottom));
+              }
+            }
+          }
+        }
+
+        if (paddingBottomForKeyboard) {
+          view.setPadding(view.getPaddingLeft(),
+                  view.getPaddingTop(),
+                  view.getPaddingRight(),
+                  paddingBottom);
+
+          if (Log.isLoggable(TAG, Log.DEBUG)) {
+            Log.d(
+                    TAG,
+                    String.format(
+                            Locale.US,
+                            "applyInsetsWhenKeyboardDisplayed. Applied padding to %s: bottom=%d}",
+                            view,
+                            marginBottom));
+          }
+        }
+
+      }
+    });
+  }
+
+  /**
+   * Determines if the keyboard is open by comparing the visible window height to the height
+   */
+  private static boolean isKeyboardOpen(View view) {
+    Rect visibleWindowBounds = new Rect();
+    view.getWindowVisibleDisplayFrame(visibleWindowBounds);
+    int visibleWindowHeight = visibleWindowBounds.height();
+    int heightDiff = view.getRootView().getHeight() - visibleWindowHeight;
+    int marginOfError = Math.round(TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 100, view.getResources().getDisplayMetrics()));
```

Comment:
Can we move this is a dimension resource? Easier to maintain and we can vary on API level/resource qualifiers if needed